src/oracle/oracleDataController.ts

```diff
@@ -0,0 +1,40 @@
+import { Controller, Get, Route, Tags } from "tsoa";
+import { getLatestSubmission, getLatestSubmissionForEachOracle } from "./oracleDataService";
+import { getPolkaBtc } from "../common/polkaBtc";
+import { OracleStatus } from "./oracleModel";
+
+interface CachedOracleData {
+    offlineCutoffDatetime: Date;
+    feed: string;
+    namesMap: Map<string, string>
+}
+
+@Tags("stats")
+@Route("oracle")
+export class OracleDataController extends Controller {
+    cachedOracleData: Promise<CachedOracleData> = (async () => {
+        const polkabtc = await getPolkaBtc();
+        const maxDelay = await polkabtc.oracle.getMaxDelay();
+        const feed = await polkabtc.oracle.getFeed();
+        const namesMap = await polkabtc.oracle.getSourcesById();
+        const offlineCutoffDatetime = new Date(Date.now() - maxDelay);
```

Comment:
Since this is ran on startup, won't this treat every oracle as online provided they submitted at least once no earlier than an hour before `stats` was launched? And then fail to pick up any oracle that goes offline subsequently.
Provided I'm not missing something, seems like the `maxDelay` is what should be cached, and then on request `Date.now()` used to calculate the cutoff _at the time of the request_, thus giving an up-to-date judgement on whether any oracle is offline at the time of the request.